src/Controls/src/Core/Platform/iOS/ControlsModalWrapper.cs

```diff
@@ -135,7 +135,11 @@ public override bool ShouldAutorotateToInterfaceOrientation(UIInterfaceOrientati
 		public override void ViewDidLayoutSubviews()
 		{
 			base.ViewDidLayoutSubviews();
-			_modal?.PlatformArrange(new Rect(0, 0, View!.Bounds.Width, View.Bounds.Height));
+			
+			if (_modal?.VirtualView is Page)
```

Comment:
This type of timing-sensitive crash can be difficult to reproduce in tests, but could create a test validating this?